cmd/kubeadm/app/cmd/token.go

```diff
@@ -130,24 +148,26 @@ func NewCmdTokenGenerate(out io.Writer) *cobra.Command {
 }
 
 // RunCreateToken generates a new bootstrap token and stores it as a secret on the server.
-func RunCreateToken(out io.Writer, cmd *cobra.Command, tokenDuration time.Duration, token string) error {
-	client, err := kubemaster.CreateClientFromFile(path.Join(kubeadmapi.GlobalEnvParams.KubernetesDir, kubeconfig.AdminKubeConfigFileName))
-	if err != nil {
-		return err
-	}
+func RunCreateToken(out io.Writer, client *clientset.Clientset, token string, tokenDuration time.Duration, usages []string) error {
 
-	parsedID, parsedSecret, err := tokenutil.ParseToken(token)
+	td := &kubeadmapi.TokenDiscovery{}
```

Comment:
Why don't you try to create the token using `td.ID, td.Secret, err = tokenutil.ParseToken(token)`, and if it returns an err, you then try to create the token using generate.  Too many checks, you could make it read more clearly.